src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2Projections.cs

```diff
@@ -0,0 +1,479 @@
+                            address.PostalCode = message.Message.PostalCode;
+                            UpdateVersionTimestamp(address, message.Message.Provenance.Timestamp);
+                        },
+                        ct);
+                }
+            });
+
+            When<Envelope<AddressHouseNumberWasCorrectedV2>>(async (context, message, ct) =>
+            {
+                await context.FindAndUpdateAddressDetail(
+                    message.Message.AddressPersistentLocalId,
+                    address =>
+                    {
+                        address.HouseNumber = message.Message.HouseNumber;
+                        UpdateVersionTimestamp(address, message.Message.Provenance.Timestamp);
+                    },
+                    ct,
+                    updateHouseNumberLabelsAfterAddressUpdate: true);
+
+                foreach (var boxNumberPersistentLocalId in message.Message.BoxNumberPersistentLocalIds)
+                {
+                    await context.FindAndUpdateAddressDetail(
+                        boxNumberPersistentLocalId,
+                        address =>
+                        {
+                            address.HouseNumber = message.Message.HouseNumber;
+                            UpdateVersionTimestamp(address, message.Message.Provenance.Timestamp);
+                        },
+                        ct,
+                        updateHouseNumberLabelsAfterAddressUpdate: true);
+                }
+            });
+
+            When<Envelope<AddressBoxNumberWasCorrectedV2>>(async (context, message, ct) =>
+            {
+                await context.FindAndUpdateAddressDetail(
+                    message.Message.AddressPersistentLocalId,
+                    address =>
+                    {
+                        address.BoxNumber = message.Message.BoxNumber;
+                        UpdateVersionTimestamp(address, message.Message.Provenance.Timestamp);
+                    },
+                    ct);
+            });
+
+            When<Envelope<AddressPositionWasChanged>>(async (context, message, ct) =>
+            {
+                await context.FindAndUpdateAddressDetail(
+                    message.Message.AddressPersistentLocalId,
+                    address =>
+                    {
+                        address.PositionMethod = ConvertGeometryMethodToString(message.Message.GeometryMethod);
+                        address.PositionSpecification = ConvertGeometrySpecificationToString(message.Message.GeometrySpecification);
+                        address.SetPosition(ParsePosition(message.Message.ExtendedWkbGeometry));
+                        UpdateVersionTimestamp(address, message.Message.Provenance.Timestamp);
+                    },
+                    ct,
+                    updateHouseNumberLabelsBeforeAddressUpdate: true,
+                    updateHouseNumberLabelsAfterAddressUpdate: true,
+                    allowUpdateRemovedAddress: true);
```

Comment:
We added allowUpdateRemovedAddress: true based upon how this was used in the V1 projections